frontend/public/components/overview/index.tsx

```diff
@@ -495,9 +495,15 @@ class OverviewMainContent_ extends React.Component<OverviewMainContentProps, Ove
       // OverviewHeading doesn't keep the value in state.
       this.setState({ filterValue: '' });
     }
+
+    // Fetch new metrics when the namespace changes.
+    if (namespace !== prevProps.namespace) {
+      clearInterval(this.metricsInterval);
+      this.fetchMetrics();
+    }
   }
 
-  fetchMetrics(): void {
+  fetchMetrics = (): void => {
```

Comment:
Using fat arrow to bind to this since we pass `this.fetchMetrics` directly to `setTimeout` below